compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/java/KSAnnotationJavaImpl.kt

```diff
@@ -67,15 +76,42 @@ class KSAnnotationJavaImpl private constructor(val psi: PsiAnnotation) : KSAnnot
         presentValueArguments.plus(argumentsFromDefault)
     }
 
-    private fun nameValuePairToKSAnnotation(annotationConstructor: ClassConstructorDescriptor?, nameIndex: Int, value: PsiAnnotationMemberValue?): KSValueArgument {
-        return KSValueArgumentJavaImpl.getCached(
-                annotationConstructor?.valueParameters?.get(nameIndex)?.name?.let { KSNameImpl.getCached(it.asString()) },
-                calcValue(value)
-        )
-    }
-
     private fun calcValue(value: PsiAnnotationMemberValue?): Any? {
-        return value?.let { JavaPsiFacade.getInstance(value.project).constantEvaluationHelper.computeConstantExpression(value) }
+        if (value is PsiAnnotation) {
+            return getCached(value)
+        }
+        val result = when(value) {
+            is PsiReference -> value.resolve()?.let { resolved ->
+                JavaPsiFacade.getInstance(value.project).constantEvaluationHelper.computeConstantExpression(value) ?: resolved
+            }
+            else -> value?.let { JavaPsiFacade.getInstance(value.project).constantEvaluationHelper.computeConstantExpression(value) }
+        }
+        return when(result) {
+            is PsiType -> {
```

Comment:
This seems to be only possible through `PsiReference.resolve()`, will it be better to merge it into the previous `when` statement?

Reply:
i tested it but we also hit this code path for this test:
```
@KotlinAnnotation(
    stringArray = {"j-a", "j-b", null, "j-c"},
    classArray = {Object.class, List.class}
)
```
for the `classArray` values.

Similarly, the `argCLs` types in `annotationValue`. 
That test wasn't failing before because the output just compares the toString, not the type of the returned value.